includes/class-freemius.php

```diff
@@ -18800,18 +18800,34 @@ private function add_submenu_items() {
                         }
                     }
 
-                    // Add upgrade/pricing page.
-                    $this->add_submenu_item(
-                        $pricing_cta_text . '&nbsp;&nbsp;' . ( is_rtl() ? $this->get_text_x_inline( '&#x2190;', 'ASCII arrow left icon', 'symbol_arrow-left' ) : $this->get_text_x_inline( '&#x27a4;', 'ASCII arrow right icon', 'symbol_arrow-right' ) ),
-                        array( &$this, '_pricing_page_render' ),
-                        $this->get_plugin_name() . ' &ndash; ' . $this->get_text_x_inline( 'Pricing', 'noun', 'pricing' ),
-                        'manage_options',
-                        'pricing',
-                        'Freemius::_clean_admin_content_section',
-                        WP_FS__LOWEST_PRIORITY,
-                        ( $add_submenu_items && $show_pricing ),
-                        $pricing_class
-                    );
+                    $custom_pricing_url        = $this->apply_filters( 'pricing_url', null );

```

Comment:
@fajardoleo Could you please refactor the `pricing_url` method and extract the logic to apply the filter in a private function

```php
function pricing_url(...) {
  $url = ...;

  return $this->get_pricing_url_with_filter( $url );
}

private function get_pricing_url_with_filter( $url ) {
  return $this->apply_filters( 'pricing_url', $url );
}
```

Then here, instead of calling the filter ad-hoc, you can do

```php
$custom_pricing_url = $this->get_pricing_url_with_filter( null );
```

This will result in a single source of truth where the filter `pricing_url` is applied so it less error prone. Thanks.